dev/documentation/en/utilities-hooks.md

```diff
@@ -0,0 +1,159 @@
+
+The hooks in the list below are available for any entity that has been converted to a [schema and service class](https://docs.pkp.sfu.ca/dev/documentation/en/architecture-entities).
+
+| Hook | Usage |
+| --- | --- |
+| `Schema::get::<schema>` | Called when an entity's schema is being loaded. Use this to modify the properties of an object. For example, use `Schema::get::publication` to modify the properties of a `Publication`. |
+| `<entity>::validate` | Validate an object before it is added or edited. For example, use `Publication::validate` to validate changes to a `Publication`. |
+| `<entity>::add` | Called when an object is added to the database. |
+| `<entity>::edit` | Called when an object is edited. |
+| `<entity>::delete::before` | Called before an object is deleted from the database. |
+| `<entity>::delete` | Called after an object is deleted from the database. |
+| `<entity>::getProperties` | Called when an object is converted to an associative array and returned in the REST API. |
+| `<entity>::getMany::queryObject` | Called when the query object is compiled in order to get a collection of objects from the database. Often used with `<entity>::getMany::getQueryBuilder`. |
+| `<entity>::getMany::getQueryBuilder` | Called when a Service class's `getMany` method is used to get a collection of objects from the database. Often used with `<entity>::getMany::queryObject`. |
+
+Some entities provide hooks for additional actions taken in the Service class.
+
+| Hook | Usage |
+| --- | --- |
+| `Publication::publish::before` | Called before a `Publication` is published. |
+| `Publication::publish` | Called after a `Publication` is published.
+| `Publication::unpublish::before` | Called before a `Publication` is unpublished. |
+| `Publication::unpublish` | Called after a `Publication` is unpublished.
+| `Publication::validatePublish` | Validate a `Publication` before it is published. Use this to ensure a publication has all the required data before publishing. |
+| `Publication::version` | Called when a new version is created for a `Publication`. |
+| `Submission::updateStatus` | Called when a `Submission` updates its `status` property. |
+| `EmailTemplate::restoreDefaults` | Called when the email templates are being restored to their defaults. |
+
+The hooks in the list below are available for any entity that has not been converted to the schema and service class. These entities use a DAO to define the properties of the model.
+
+| Hook | Usage |
+| --- | --- |
+| `<dao>::_fromRow` | Map the rows retrieved from a database to a [DataObject](./architecture-entities#dataobject-class). For example, use `SectionDAO::_fromRow` to modify a `Section`. |
+| `<dao>::getAdditionalFieldNames` | Modify the properties of an entity that are stored in its settings table. For example, use `SectionDAO::getAdditionalFieldNames` to modify the properties of a `Section`. |
```

Comment:
I *think* we want to deprecate this one in favour of the SchemaDAO tools. Maybe continue to document it, but add a note?

Reply:
Is the note just above the table not enough? https://github.com/pkp/pkp-docs/pull/391/files/85634d140e46173021889b4f89b5e6c79bad61ee#diff-e547b1fb1639b51b4e013ba2f2555487R141

> The hooks in the list below are available for any entity that has not been converted to the schema and service class. These entities use a DAO to define the properties of the model.

Reply:
Whoops, yes, I totally missed that :)